modules/perception/onboard/component/trafficlights_perception_component.cc

```diff
@@ -83,6 +86,8 @@ static int GetGpuId(
 }
 
 bool TrafficLightsPerceptionComponent::Init() {
+  std::shared_ptr<camera::CameraFrame> p(new camera::CameraFrame);
+  frame_.swap(p);
```

Comment:
why not make_shared?

Reply:
make_shared may cause Eigen error  @lfcarol

Reply:
or just frame_ = std::shared_ptr<camera::CameraFrame> (new camera::CameraFrame);?
or frame_.reset(new camera::CameraFrame );
:)

Reply:
It's a bit awkard to instantiate a new shared pointer and swap with an existing one. Why not reset it to the existing value or instantialize it to the existing value?

Reply:
Please work with @changsh726 and @lfcarol  for an efficient use of it.

Reply:
> make_shared may cause Eigen error @lfcarol

Fix it by "frame_.reset(new camera::CameraFrame);" , "make_shared" will case crash.

Reply:
> Please work with @changsh726 and @lfcarol for an efficient use of it.

OK